### PR TITLE
feat: adding multi dimensional flavors example

### PR DESCRIPTION
**Pull request by erickzanardo**
## Description

Adds a simple sample, very similar to the `flavors` one, but which includes multi dimensional flavors.

Since this is a feature only present on android, this sample has only the android folder.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 🧪 Tests
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by bryanoltman — APPROVED** [review]

File: multi_dimensional_flavors/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,125 @@
+import 'package:flutter/material.dart';
```
Comment (felangel) [inline multi_dimensional_flavors/lib/main.dart:1]: Minor but we could reduce the amount of code here by using the template generated from `flutter create --empty .`

File: multi_dimensional_flavors/shorebird.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# This file is used to configure the Shorebird updater used by your app.
+# Learn more at https://docs.shorebird.dev
+# This file should be checked into version control.
+
+# This is the unique identifier assigned to your app.
+# Your app_id is not a secret and is just used to identify your app
+# when requesting patches from Shorebird's servers.
+app_id: 5171ee3d-1a4d-4bdd-91c3-64cef78fe41c
+flavors:
+  internal: 5171ee3d-1a4d-4bdd-91c3-64cef78fe41c
```
Comment (felangel) [inline multi_dimensional_flavors/shorebird.yaml]: I don't think this is a valid configuration/app so we're unnecessarily creating an app which can never be used.

File: multi_dimensional_flavors/shorebird.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  internalGlobal: 5487a0d3-8c5a-4178-86ef-5295588e89bd
+  internalLocal: acb850fb-a9e3-45fb-86c0-705bb265d1e4
+  internalPlayStore: c8cfaff9-d94e-4767-9055-50c4f0b25ddd
+  stable: 666541bc-0eda-477d-8ad5-ec97760435e6
```
Comment (felangel) [inline multi_dimensional_flavors/shorebird.yaml]: same here

File: multi_dimensional_flavors/shorebird.yaml (hunk 2 above)
Comment (erickzanardo) [reply multi_dimensional_flavors/shorebird.yaml]: Makes sense, there is even this issue here to track this: https://github.com/shorebirdtech/shorebird/issues/1961

Good catch, I am going to remove the invalid apps from the example

**Pull request merged**